ADRs/ADR0006-replace-badger-kv-store [WITHDRAWN].md

```diff
@@ -0,0 +1,225 @@
+
+## Context and Problem Statement
+
+When applying blocks, Postgres validates the block contents. However, Badger stores key/value pairs that make up the SMT representation of transactions for each block during block application. When a block fails validation and thus application, i.e. Postgres declares the data invalid, a rollback is initiated in Postgres, but the key/value pairs of that failed transaction are still left in Badger. The problem lies in the KVStore interface. The KVStore interface does not currently expose commit handling but acts as a subordinate component of an atomic persistent context. This means that its wrapper, the persistence context, can't itself provide specific atomic guarantees.
+
+## Design Goals
+
+The goal is synchronizing the atomic application or reversal of block data between both Postgres and Badger such that we are not left with orphaned keys or rows in the KVStore or the Postgres database.
+
+## Decision Drivers
+
+* The ability to handle atomic rollbacks is crucially important, but the split between Badger and Postgres means one more variable to handle in an already complex process of advancing and rolling back state when necessary.
+* Simplifying this feature could eliminate a dependency and decreases maintenance costs.
+
+## Considered Options
+
+* Use Postgres for atomic guarantees
+  * Store in HSTORE
+  * Store in JSONB
+  * Store in Plain Table
+* Handle atomic guarantees at the KVStore
+* Manually cleaning up orphaned keys
+
+## Decision Outcome
+
+This ADR is withdrawn. The decision has been made to focus on refactoring the KVStore interface to guarantee atomic block application. That relates to exploration in Option 2 in this document.
+
+### Positive Consequences
+
+* Allows us to remove a dependency on Badger.
+* Simplifies transaction handling for savepoints and rollbacks.
+
+### Negative Consequences
+
+* Any of the three Postgres KV store implementations outlined below requires a migration to complete.
+
+```go
+// TODO(#77): Enable proper up and down migrations
+func initializeDatabase(conn *pgxpool.Conn) error {
+  // TODO: IN THIS COMMIT initialize key value store if it doesn't exist
+  // Initialize the tables if they don't already exist
+  if err := initializeAllTables(context.TODO(), conn); err != nil {
+    return fmt.Errorf("unable to initialize tables: %v", err)
+  }
+  return nil
+}
+```
+
+* Badger is highly performant and the KV store in place currently is very efficient. Changing this part of the system could mean changing back in the future in the face of performance issues.
```

Comment:
NIT: Make this the first bullet point so there's no large delimiter with the code